### DISCUSSION TITLE
Multi language monorepo

### DISCUSSION THREAD

**Discussion by NixBiks** [vercel/turborepo#194] [Help] [answered]
Hi all,

Have anybody tried building a multi language monorepo, e.g. NodeJS and python applications? Wonder if turborepo can be beneficial for managing dependencies for python services as well

**Comment by jaredpalmer** [comment, marked as answer]
It’s something I’ve considered, but right now focusing on JS. Of course, can def get around this like we do in the turbo monorepo for go  by putting package.json in the folders you want.

**Reply by NixBiks** [reply]
Yeah I can do that but I guess there are no good things to cache though. 

The time consuming part for python projects is building the virtual environments but I assume you don't want to put that in the cache? Wondering if there are any recommended settings working with python environments?

@jaredpalmer

**Reply by johannes-graeter** [reply]
Hi Jared,

thanks for this amazing project. Ia m working a lot with (modern) C++. One of its major drawbacks is that build time gets large quickly and  can really  slow down productivity for this. If I understand correctly I could hence compile my c++ project as shown in https://github.com/vercel/turborepo/blob/main/cli/package.json for the go example right? But probably most of turborepos magic will be gone then, since this will generate 1 single artifact and hence its parallelization and remote caching of mid process artifacts will be lost...
Do you see a way of integrating this with CMake, for example (or any other c++ build system) or a starting point for getting involved into this? 

Best,
Johannes
